packages/payment-detection/src/declarative.ts

```diff
@@ -12,9 +12,12 @@ import { PaymentDetectorBase } from './payment-detector-base';
  */
 export abstract class DeclarativePaymentDetectorBase<
   TExtension extends ExtensionTypes.PnAnyDeclarative.IAnyDeclarative,
-  TPaymentEventParameters extends PaymentTypes.IDeclarativePaymentEventParameters,
+  TPaymentEventParameters extends PaymentTypes.IDeclarativePaymentEventParameters<unknown>,
```

Comment:
Here is an example of a hidden mistyping that cannot be missed anymore.
More specifically, Declarative payment events are at the root of many detectors, but when events are declarative for real, they use `IIdentity` type for `from`, whereas it's `string` for overriding children (when it's not declarative)